src/alert/Pairing.ts

```diff
@@ -0,0 +1,279 @@
+import { ExtensionMessage, ExtensionMessageTarget, NetworkType, availableTransports } from '..'
+import { myWindow } from '../MockWindow'
+import { getTzip10Link } from '../utils/get-tzip10-link'
+import { isAndroid, isIOS } from '../utils/platform'
+import { desktopList, extensionList, iOSList, webList } from './wallet-lists'
+
+const defaultExtensions = [
+  'ookjlbkiijinhpmnjffcofjonbfbgaoc', // Thanos
+  'gpfndedineagiepkpinficbcbbgjoenn' // Beacon
+]
+
+export enum Platform {
+  DESKTOP,
+  IOS,
+  ANDROID
+}
+
+export enum WalletType {
+  IOS = 'ios',
+  ANDROID = 'android',
+  EXTENSION = 'extension',
+  DESKTOP = 'desktop',
+  WEB = 'web'
+}
+
+export interface AppBase {
+  name: string
+  shortName: string
+  color: string
+  logo: string
+}
+
+export interface ExtensionApp extends AppBase {
+  id: string
+  link: string
+}
+
+export interface WebApp extends AppBase {
+  links: {
+    [NetworkType.MAINNET]: string
+    [NetworkType.CARTHAGENET]?: string
+    [NetworkType.DELPHINET]?: string
+    [NetworkType.CUSTOM]?: string
+  }
+}
+
+export interface App extends AppBase {
+  universalLink: string
+  deepLink?: string
+}
+
+export interface PairingAlertWallet {
+  name: string
+  shortName?: string
+  color?: string
+  logo?: string
+  enabled: boolean
+  clickHandler(): void
+}
+
+export interface PairingAlertButton {
+  text: string
+  clickHandler(): void
+}
+
+export interface PairingAlertList {
+  title: string
+  type: WalletType
+  wallets: PairingAlertWallet[]
+}
+
+export interface PairingAlertInfo {
+  walletLists: PairingAlertList[]
+  buttons: PairingAlertButton[]
+  qrData: string
+}
+
+export type StatusUpdateHandler = (walletType: WalletType, app?: PairingAlertWallet) => void
+
+export class Pairing {
+  public static async getPlatfrom(): Promise<Platform> {
+    return isAndroid(window) ? Platform.ANDROID : isIOS(window) ? Platform.IOS : Platform.DESKTOP
+  }
+
+  public static async getPairingInfo(
+    pairingPayload: {
+      p2pSyncCode: string
+      postmessageSyncCode: string
+      preferredNetwork: NetworkType
+    },
+    statusUpdateHandler: StatusUpdateHandler,
+    platform?: Platform
+  ): Promise<PairingAlertInfo> {
+    const activePlatform = (await Pairing.getPlatfrom()) ?? platform
```

Comment:
@AndreasGassmann shouldn't the input `platform` parameter be able to override the result from `Pairing.getPlatfrom()`?

Reply:
Yes, fixed